src/harness/testcases/registration_testcase.py

```diff
@@ -543,7 +543,90 @@ def test_WINFF_FT_S_REG_13(self):
     self.assertEqual(response['response']['responseCode'], 200)
 
   @winnforum_testcase
-  def test_WINFF_FT_S_REG_19(self):
+  def test_WINNF_FT_S_REG_15(self):
+    """Invalid parameters in Array request (responseCode 103)
+
+    The response should be SUCCESS for the first device,
+    FAIL for the second, third, fourth, fifth, and sixth devices.
+    """
+
+    # Load the devices
+    device_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    device_3 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_e.json')))
+    device_4 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_f.json')))
+    device_5 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_g.json')))
+    device_6 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+
+    # meascapability has no value for all devices
+    device_1['measCapability'] = []
+    device_2['measCapability'] = []
+    device_3['measCapability'] = []
+    device_4['measCapability'] = []
+    device_5['measCapability'] = []
+    device_6['measCapability'] = []
+
+    # Device 1 Cat A
+    self.assertEqual(device_1['cbsdCategory'], 'A')
+
+    # Device 2 Cat A invalid cbsdSerialNumber - above max length of 64 octets
+    self.assertEqual(device_2['cbsdCategory'], 'A')
+    device_2['cbsdSerialNumber'] = 'a' * 65
+
+    # Device 3 Cat A invalid fccId - above max length of 19 chars
+    self.assertEqual(device_3['cbsdCategory'], 'A')
+    device_3['fccId'] = 'a' * 20
+
+    # Device 4 Cat A invalid userId - invalid char (RFC-7542 Section 2.2)
+    self.assertEqual(device_4['cbsdCategory'], 'A')
+    device_4['userId'] = '^'
+
+    # Device 5 Cat A invalid latitude - invalid type
+    self.assertEqual(device_5['cbsdCategory'], 'A')
+    device_5['installationParam']['latitude'] = 'abc'
```

Comment:
Maybe we want to test out of range (e.g. 91) since that's much more likely to happen in practice?